pageserver/src/remote_storage/local_fs.rs

```diff
@@ -154,6 +153,64 @@ impl RemoteStorage for LocalFs {
         }
     }
 
+    async fn download_range(
+        &self,
+        from: &Self::StoragePath,
+        start_inclusive: u64,
+        end_exclusive: Option<u64>,
+        to: &mut (impl io::AsyncWrite + Unpin + Send + Sync),
+    ) -> anyhow::Result<()> {
+        if let Some(end_exclusive) = end_exclusive {
+            ensure!(
+                end_exclusive > start_inclusive,
+                "Invalid range, start ({}) is bigger then end ({:?})",
+                start_inclusive,
+                end_exclusive
+            );
+            if start_inclusive == end_exclusive - 1 {
+                return Ok(());
+            }
```

Comment:
Would be nice to avoid panic if end_exclusive == 0

Reply:
Good catch, thanks, fixed.